### PR TITLE
Fix cudnn error for BatchNorm1D kernel

### PR DESCRIPTION
**Pull request by EsdeathYZH**
### PR types

Function optimization
### PR changes

OPs
### Describe

1 Motivation：
+ BatchNorm1D算子在batch size过大时（例：[136000, 16]）下报错
+ 该问题在torch中也被处理了，参考 https://github.com/pytorch/pytorch/pull/31976

2 Design：
+ 根据Batch size判断是否使用cudnn库进行计算，当batch size大于阈值(参考了https://github.com/pytorch/pytorch/pull/31976 设为131070)时使用native kernel
+ 为了不影响现有大部分模型的精度，暂时只在报错的case下使用新增的kernel（BatchNorm1D, batchsize > [THRESHOLD]）

3 Evaluation：
+ TODO

### CODE REVIEW

File: paddle/phi/kernels/gpu/batch_norm_kernel.cu [hunk 1]
```diff
@@ -446,90 +446,78 @@ void BatchNormKernel(const Context &ctx,
...
+//         static_cast<void *>(variance_out->template mutable_data<
+//                             BatchNormParamType<T>>(ctx.GetPlace())),
+//         epsilon,
+//         static_cast<void *>(
+//             saved_mean->template mutable_data<BatchNormParamType<T>>(
+//                 ctx.GetPlace())),
+//         static_cast<void *>(saved_variance->template mutable_data<
+//                             BatchNormParamType<T>>(ctx.GetPlace()))));
+#else
+      const bool use_native_kernel = (x_dims.size() == 2 && N >= 131070);
```
Comment (zkh2016) [inline paddle/phi/kernels/gpu/batch_norm_kernel.cu]: 131070可以放到一个const变量里，或者加个注释说明下这个阈值。

Comment (EsdeathYZH) [reply paddle/phi/kernels/gpu/batch_norm_kernel.cu]: done

**Review by zkh2016 — DISMISSED** [review]

**Review by ceci3 — DISMISSED** [review]
LGTM

**Review by zkh2016 — APPROVED** [review]

**Review by JamesLim-sy — APPROVED** [review]
LGTM. 能否在comment区域补充下，之前测试case的性能以及与竞品的对比：
- 是否可以跑大bs；
- 大bs下的性能对比。

**Comment by EsdeathYZH** [comment]
> LGTM. 能否在comment区域补充下，之前测试case的性能以及与竞品的对比：
> 
> * 是否可以跑大bs；
> * 大bs下的性能对比。

这个PR主要是先处理掉大batch下报错的问题，有关性能方面的进一步优化在 https://github.com/PaddlePaddle/Paddle/pull/43530 中进行了详细的描述 @JamesLim-sy

**Review by chenwhql — APPROVED** [review]

**Review by ceci3 — APPROVED** [review]
